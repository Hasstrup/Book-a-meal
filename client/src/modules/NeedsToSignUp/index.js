import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { SignUpUser, LogInUser } from '../../actions/users';
import { DispatchNotification } from '../../actionTypes/misc';
import './index.scss';

const SignUpComponent = ({ renderLogin, handleClick, handleSubmit, handleChange }) => (
  <div className="sign-up-user-form">
    <div className="render-form-single">
      <label htmlFor="email" className="render-form-label"> Email </label>
      <input onChange={handleChange} type="email" required name="email" className="render-form-input" />
    </div>
    <div className="render-form-single">
      <label htmlFor="password" > Password </label>
      <input onChange={handleChange} type="password" required className="render-form-input" name="password" />
    </div>
    { !renderLogin &&
    <Fragment>
      <div className="render-form-single">
        <label htmlFor="name" className="render-form-label"> Full Name</label>
        <input onChange={handleChange} type="text" required className="render-form-input" name="name" />
      </div>
      <div className="render-form-single">
        <label htmlFor="username" className="render-form-label"> Username </label>
        <input onChange={handleChange} required className="render-form-input" name="username" />
      </div>
    </Fragment>
    }
    <button onClick={handleSubmit}> Submit </button>
    <p className="__user_already" onClick={handleClick}>{ renderLogin ? 'Create new account' : 'I already have an account '}</p>
  </div>
);

class AuthenticateUserForm extends Component {
   state = { renderLogin: false, preamble: "Holla. Let's get you started" };
    handleChange = e => this.setState({ [e.target.getAttribute('name')]: e.target.value });

    
    handleSubmit = () => {
      const { email, password, username, name } = this.state;
      const filterThrough = obj => Object.values(obj).filter(item => !item || !item.length).length;
      if (this.state.renderLogin && !filterThrough({ email, password })) return this.props.dispatch(LogInUser({ email, password })(this.props.history));
      if (!filterThrough({ name, password, username, email })) return this.props.dispatch(SignUpUser({ email, password, firstname: name.split(' ')[0], username })(this.props.history));
      this.props.dispatch(DispatchNotification('Hey, please fill in the required fields'));
    }

    render = () => (
      <div className="render-auth-modal" style={{ display: this.props.requiresPermission ? 'flex' : 'none' }}>
        <div className="auth-modal-backdrop" />
        <div className="auth-modal-main">
          <div className="__auth-modal-house">
            <p className="auth-preamble">{this.state.preamble}</p>
            <p className="auth-preamble-text">{this.props.preamble || ''}</p>
          </div>
          <div className="__auth-modal-main">
            <SignUpComponent
              handleClick={() => this.setState({ renderLogin: !this.state.renderLogin, preamble: (this.state.renderLogin ? "Holla. Let's get you started" : 'Hungry again? Nice.') })}
              renderLogin={this.state.renderLogin}
              handleChange={this.handleChange}
              handleSubmit={this.handleSubmit}
            />
          </div>
        </div>
      </div>
    )
}

const mapStateToProps = state => ({
    requiresPermission: state.users.requiresPermission
})

export default connect(mapStateToProps)(withRouter(AuthenticateUserForm));

