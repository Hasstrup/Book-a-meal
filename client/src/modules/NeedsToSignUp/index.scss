.render-auth-modal {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1000;
    display: flex;
    background-color: rgba(253, 254, 254, 0.9);
    .auth-modal-backdrop {
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
    }
    .auth-modal-main {
        display: flex; 
        height: 70%;
        width: 60%;
        background-color: white;
        border-radius: 5px;
        align-self: center;
        margin: 0 auto;
        position: relative; 
        right: 35%;
        box-shadow: 1px 1px 10px 1px #F2F3F4;
        flex-flow: column nowrap;
        align-items: center;
        .__auth-modal-house {
            height: 10%;
            width: 100%;
            .auth-preamble {
                color: #5A5A5A;
                text-align: center;
                font: {
                    size: 25px;
                    weight: bold;
                    family: gotham;
                }
            }
        }
        .__auth-modal-main {
            height: 90%;
            width: 100%;
            display: flex; 
            flex-flow: column nowrap;
            align-items: center;
            .sign-up-user-form {
                display: flex;
                height: 100%;
                width: 50%;
                align-self: center;
                flex-flow: column nowrap;
                justify-content: space-evenly;
                .__user_already {
                    font-size: 13px; 
                    font-family: apercu;
                    color: #BDC3C7;
                    text-align: center;
                    &:hover {
                        cursor: pointer;
                        text-decoration: underline;
                    }
                }
                .render-form-single {
                    .render-form-input {
                        display: block;
                        height: 40px;
                        width: 100%;
                        border: none;
                        border-bottom: 0.5px solid #E5E7E9;
                        font-family: apercu;
                        font-size: 14px;
                        &:focus {
                            outline: none;
                        }
                    }
                    label {
                        color: #BDC3C7;
                    }
                    font: {
                        size: 16px;
                        family: apercu;
                    }
                }
                button {
                    background-color: #ff183f;
                    color: white;
                    font-family: gotham;
                    font-size: 18px;
                    height: 50px !important; 
                    width: 290px;
                    border: none;
                    border-radius: 3px;
                    &:hover {
                        cursor: pointer;
                    }
                }
            }
        }
    }
}