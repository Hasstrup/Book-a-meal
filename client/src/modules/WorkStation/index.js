import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DispatchNotification } from '../../actionTypes/misc';
import RenderWorkStationMain from './components/workstation-main';
import RenderKitchenAndUserBio from './components/UserBio';
import '../styles/profile.scss';
import { FetchUser } from '../../actions/users';
import { ProcessIndicatorLg } from '../../mixins/ProcessIndicator';
import KitchenActions from '../../actions/kitchens';
import MealActions from '../../actions/meals';
import utils from './utils';

// TODO: you might want to destructure from the exports itself.
const { RenderMealForm, GetMealInformation, HideMealForm } = utils;
const { createNewMeal, fetchAllMealsBelongingToUser } = MealActions;
const { SetUpNewKitchen } = KitchenActions;

class WorkStationContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

   componentDidMount = () => {
     if (!this.props.user) return this.props.dispatch(FetchUser());
     this.props.dispatch(fetchAllMealsBelongingToUser()); // just so that this updates with page transitions
   }


 handleSubmit = (key) => {
   switch (key) {
     case 1:
       return this.handleNewKitchen();

     case 2:
       return this.handleNewMeal();

     default:
       return null;
   }
 }

 // TODO: You might need to extract this so it's accessible to all your components
  /**
 * @description Utility function that checks if input passes a validity check. Checks that none of the fields are 
 * default values provided on render
 * @returns {boolean} the result of auth test
 * @private
 */
 validMealInput = (meal) => {
   const target = Object.values(meal);
   // doing this because you want a valid price
   const BlackList = ['Meal name', 'description', 'Price', ''];
   return target.reduce((acccumulator, currentValue) => {
     if (BlackList.includes(currentValue)) {
       acccumulator = false;
     }
     /* eslint-disable-next-line no-restricted-globals, radix */
     if (meal.price && meal.price.length && isNaN(parseInt(meal.price, 10))) {
       acccumulator = false;
     }
     return acccumulator;
   }, true);
 }

  /**
 * @description Renders the meal form for the user
 * @returns {null}
 * @private
 */
 handleRenderMealForm = () => {
   RenderMealForm();
   this.state.wantsToAddMeal = true; // doing this to avoid a rerender of the dom;
 }

/**
 * @description This either renders the meal form or dispatches the set up meal 
 * variant on the number of times the newMeal button is clicked
 * @returns {null}
 * @private
 */
 handleNewMeal = () => {
   if (!this.state.wantsToAddMeal) return this.handleRenderMealForm();
   if (!this.validMealInput(GetMealInformation())) return this.props.dispatch(DispatchNotification(`Hey ${this.props.user.firstname}, you need to fill in every field, correctly too :)`));
   this.props.dispatch(createNewMeal(GetMealInformation())(this.resetMealForm)); // reset meal should be bound;
 }

 /**
 * @description Hides the meal form after a successful set up of the meal
 * resets the form data to default values
 * @returns {null}
 * @private
 */
 resetMealForm = () => {
   HideMealForm();
   this.state.wantsToAddMeal = false; // avoiding a rerender here as well
 }

  /**
 * @description Sets up a new kitchen for the user
 * @returns {function} the call to the dispatch function// might not be neccessary to note in this case
 * @private
 */
   handleNewKitchen = () => {
     const NewKitchenName = document.getElementsByClassName('new-kitchen-d')[0].innerText;
     const NewKitchenDescription = document.getElementsByClassName('new-kitchen-d')[1].innerText;
     // check to make sure the value is not the placeholder and not empty;
     if (!NewKitchenName.length || !NewKitchenDescription.length || NewKitchenName === 'Kitchen Name' || NewKitchenDescription === 'Description') {
       return this.props.dispatch(DispatchNotification('Sorry, you need to fill in every field'));
     }
     return this.props.dispatch(SetUpNewKitchen({ name: NewKitchenName, description: NewKitchenDescription }));
   }


    render = () => (
      <div>
        { this.props.user ?
          <div className="main-profile-body">
            <RenderKitchenAndUserBio user={this.props.user} handleChange={this.handleChange} handleSubmit={this.handleSubmit} />
            <RenderWorkStationMain kitchen={this.props.user.Kitchen} handleSubmit={this.handleSubmit} meals={this.props.meals} />
          </div>
        : <ProcessIndicatorLg />
      }
      </div>
    )
}

const mapStateToProps = state => ({
  user: state.users.target,
  meals: state.meals.belongsToUser
});

export default connect(mapStateToProps)(WorkStationContainer);
