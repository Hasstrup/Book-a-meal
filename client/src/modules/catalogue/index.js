import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import '../styles/catalogue.scss';
import { MainCatalogue } from '../../mixins/cards/singleMenu';
import { FetchCatalogue } from '../../actions/menus/';


class Catalogue extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    menus: PropTypes.arrayOf(PropTypes.object).isRequired,
    history: PropTypes.object.isRequired
  }

  componentDidMount = () => {
    if (!this.props.menus) return this.fetchMenus();
  }

  fetchMenus = () => this.props.dispatch(FetchCatalogue(this.props.history));

  render = () => (
    <div className="main-body">
      <CatalogueFirstRow />
      <Selectors />
      <CatalogueGridMain history={this.props.history} menus={this.props.menus || []} />
    </div>
  );
}

/*= ========== Components ============================ */

const CatalogueFirstRow = () => (
  <div className="introductory-grid">
    <div className="side-one-IT">
      <div className="side-one-child-1">
        <p className="side-one-IT-cat"> Catalogue </p>
        <div className="side-one-trending"> Today</div>
      </div>
      <p className="side-one-IT-option"> <span> 237 </span> Menus </p>
      <p className="side-one-IT-option"> <span> 1034</span> Meal options </p>
      <p className="side-one-IT-option"> <span> 101 </span> Vendors</p>
    </div>

    <div className="side-two-IT">
      <p className="side-two-IT-info">
                Sharing a menu is super easy! There's no restrictions
                whatsoever, all you have to do set up your kitchen/catering
                service, add some meal options, set up a menu and we're good
                to go .
      </p>
      <p className="side-two-IT-info" id="side-two-share-button"> Share your menu now</p>
    </div>
  </div>
);


const Selectors = () => (
  <div className="options-grid">
    <p> Top </p>
    <p> Featured </p>
    <p> All </p>
    <p> Favourites </p>
    <p> Promote </p>
  </div>
);


const CatalogueGridMain = props => (
  <div className="main-items-grid">
    { /* remember to get some mock || main data to fill in this place */ }
    { MainCatalogue(props.menus)(props) }
  </div>
);

const mapStateToProps = state => ({
  menus: state.menus.catalog
});
// these should be refactored after the application is completed right
export default connect(mapStateToProps)(Catalogue);

