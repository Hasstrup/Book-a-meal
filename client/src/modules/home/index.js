import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../styles/index.scss';
import AnimatingBannerComponent from './components/banner';
import ImageFloatingComponent from './components/floater';
import LoginFormComponent from './components/auth';
import { SignUpUser, LogInUser, GetLoggedInUser } from '../../actions/users/';
/* eslint-disable */


export class HomeContainer extends Component {
  constructor(props) {
    super(props);
    if(this.props.user) return this.props.history.push('/catalogue');
    
  }
  componentDidMount = () => {
    GetLoggedInUser();
  }
  
  render = () => (<HomeComponent history={this.props.history} dispatch={this.props.dispatch} />)
}

export const HomeComponent = ({ history, dispatch }) => (
  <div>
  <AnimatingBannerComponent />
  <LoginFormComponent logInUser={(body) => dispatch(LogInUser(body)(history))} createUser={(body) => dispatch(SignUpUser(body)(history))}/>
  </div>
)

const mapStateToProps = (state) => ({
    user: state.users.current
})

export default connect(mapStateToProps)(HomeContainer);
