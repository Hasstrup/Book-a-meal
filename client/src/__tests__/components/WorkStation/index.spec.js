import React from 'react';
import { shallow } from 'enzyme';
import { stub } from 'sinon';
import defaultProps from '../../helpers';
import { WorkStationContainer } from '../../../modules/WorkStation';

let wrapper;
const menu = {
  name: 'Test Menu',
  description: 'This is a test menu',
  kitchen: {
    name: 'This is great',
    description: 'This is my test menu',
    id: 3,
    MenuofTheDay: null
  },
  meals: [{
    name: 'This is a menu',
    description: '',
    price: 5000,
    kitchenId: 2
  }]
};

const user = {
  firstname: 'Hasstru',
  lastname: 'Ezekiel',
  email: 'hass@mail.com',
  kitchen: menu.kitchen
};
describe('WorkStation Container', () => {
  it('should match snapshot on shallow render', () => {
    wrapper = shallow(<WorkStationContainer {...defaultProps} user={user} kitchen={menu.kitchen} meals={menu.meals} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Workstation methods', () => {
    it('validMealInput should return true or false for a valid meal input', () => {
      const meal = { name: 'Test Meal', description: 'description123', price: 200, };
      const { validMealInput } = wrapper.instance();
      expect(validMealInput(meal)).toEqual(true);
      meal.price = 'string';
      expect(validMealInput(meal)).toEqual(false);
    });

    it('handleSubmit should render the right forms based on the key passed', () => {
      const { handleSubmit } = wrapper.instance();
      const KitchenStub = stub(wrapper.instance(), 'handleNewKitchen');
      const MealStub = stub(wrapper.instance(), 'handleNewMeal');
      handleSubmit(1);
      expect(KitchenStub.called).toEqual(true);
      handleSubmit(2);
      expect(MealStub.called).toEqual(true);
      KitchenStub.restore();
      MealStub.restore();
    });
  });

  it('handleNewMeal should show the render meal form', () => {
    const { handleNewMeal } = wrapper.instance();
    handleNewMeal();
    expect(wrapper.instance().state.wantsToAddMeal).toEqual(true);
  });

  it('should render successfully on deep mounting', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
