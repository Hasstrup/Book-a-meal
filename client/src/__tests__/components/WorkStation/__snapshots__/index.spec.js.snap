// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkStation Container should match snapshot on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkStationContainer
    dispatch={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    kitchen={
      Object {
        "MenuofTheDay": null,
        "description": "This is my test menu",
        "id": 3,
        "name": "This is great",
      }
    }
    meals={
      Array [
        Object {
          "description": "",
          "kitchenId": 2,
          "name": "This is a menu",
          "price": 5000,
        },
      ]
    }
    user={
      Object {
        "email": "hass@mail.com",
        "firstname": "Hasstru",
        "kitchen": Object {
          "MenuofTheDay": null,
          "description": "This is my test menu",
          "id": 3,
          "name": "This is great",
        },
        "lastname": "Ezekiel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="main-profile-body"
        style={Object {}}
      >
        <Unknown
          handleSubmit={[Function]}
          user={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
        />
        <Unknown
          callback={[Function]}
          determineRenderBy={
            Array [
              "name",
              "description",
              "id",
              "MenuofTheDay",
            ]
          }
          emptyContainerStyle={
            Object {
              "bottom": "-20vh",
              "height": "10%",
              "position": "relative",
              "width": "100%",
            }
          }
          handleSubmit={[Function]}
          image="empty"
          kitchen={
            Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            }
          }
          meals={
            Array [
              Object {
                "description": "",
                "kitchenId": 2,
                "name": "This is a menu",
                "price": 5000,
              },
            ]
          }
          text="Looks like you need to set up a kitchen"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            handleSubmit={[Function]}
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />,
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              }
            }
            handleSubmit={[Function]}
            image="empty"
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            meals={
              Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ]
            }
            text="Looks like you need to set up a kitchen"
          />,
        ],
        "className": "main-profile-body",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": undefined,
            "handleSubmit": [Function],
            "user": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callback": [Function],
            "determineRenderBy": Array [
              "name",
              "description",
              "id",
              "MenuofTheDay",
            ],
            "emptyContainerStyle": Object {
              "bottom": "-20vh",
              "height": "10%",
              "position": "relative",
              "width": "100%",
            },
            "handleSubmit": [Function],
            "image": "empty",
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "meals": Array [
              Object {
                "description": "",
                "kitchenId": 2,
                "name": "This is a menu",
                "price": 5000,
              },
            ],
            "text": "Looks like you need to set up a kitchen",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="main-profile-body"
          style={Object {}}
        >
          <Unknown
            handleSubmit={[Function]}
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              }
            }
            handleSubmit={[Function]}
            image="empty"
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            meals={
              Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ]
            }
            text="Looks like you need to set up a kitchen"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              handleSubmit={[Function]}
              user={
                Object {
                  "email": "hass@mail.com",
                  "firstname": "Hasstru",
                  "kitchen": Object {
                    "MenuofTheDay": null,
                    "description": "This is my test menu",
                    "id": 3,
                    "name": "This is great",
                  },
                  "lastname": "Ezekiel",
                }
              }
            />,
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  "name",
                  "description",
                  "id",
                  "MenuofTheDay",
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-20vh",
                  "height": "10%",
                  "position": "relative",
                  "width": "100%",
                }
              }
              handleSubmit={[Function]}
              image="empty"
              kitchen={
                Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                }
              }
              meals={
                Array [
                  Object {
                    "description": "",
                    "kitchenId": 2,
                    "name": "This is a menu",
                    "price": 5000,
                  },
                ]
              }
              text="Looks like you need to set up a kitchen"
            />,
          ],
          "className": "main-profile-body",
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": undefined,
              "handleSubmit": [Function],
              "user": Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "callback": [Function],
              "determineRenderBy": Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ],
              "emptyContainerStyle": Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              },
              "handleSubmit": [Function],
              "image": "empty",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "meals": Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ],
              "text": "Looks like you need to set up a kitchen",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WorkStation Container should render successfully on deep mounting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkStationContainer
    dispatch={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    kitchen={
      Object {
        "MenuofTheDay": null,
        "description": "This is my test menu",
        "id": 3,
        "name": "This is great",
      }
    }
    meals={
      Array [
        Object {
          "description": "",
          "kitchenId": 2,
          "name": "This is a menu",
          "price": 5000,
        },
      ]
    }
    user={
      Object {
        "email": "hass@mail.com",
        "firstname": "Hasstru",
        "kitchen": Object {
          "MenuofTheDay": null,
          "description": "This is my test menu",
          "id": 3,
          "name": "This is great",
        },
        "lastname": "Ezekiel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="main-profile-body"
        style={Object {}}
      >
        <Unknown
          handleSubmit={[Function]}
          user={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
        />
        <Unknown
          callback={[Function]}
          determineRenderBy={
            Array [
              "name",
              "description",
              "id",
              "MenuofTheDay",
            ]
          }
          emptyContainerStyle={
            Object {
              "bottom": "-20vh",
              "height": "10%",
              "position": "relative",
              "width": "100%",
            }
          }
          handleSubmit={[Function]}
          image="empty"
          kitchen={
            Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            }
          }
          meals={
            Array [
              Object {
                "description": "",
                "kitchenId": 2,
                "name": "This is a menu",
                "price": 5000,
              },
            ]
          }
          text="Looks like you need to set up a kitchen"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            handleSubmit={[Function]}
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />,
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              }
            }
            handleSubmit={[Function]}
            image="empty"
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            meals={
              Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ]
            }
            text="Looks like you need to set up a kitchen"
          />,
        ],
        "className": "main-profile-body",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": undefined,
            "handleSubmit": [Function],
            "user": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callback": [Function],
            "determineRenderBy": Array [
              "name",
              "description",
              "id",
              "MenuofTheDay",
            ],
            "emptyContainerStyle": Object {
              "bottom": "-20vh",
              "height": "10%",
              "position": "relative",
              "width": "100%",
            },
            "handleSubmit": [Function],
            "image": "empty",
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "meals": Array [
              Object {
                "description": "",
                "kitchenId": 2,
                "name": "This is a menu",
                "price": 5000,
              },
            ],
            "text": "Looks like you need to set up a kitchen",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="main-profile-body"
          style={Object {}}
        >
          <Unknown
            handleSubmit={[Function]}
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              }
            }
            handleSubmit={[Function]}
            image="empty"
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            meals={
              Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ]
            }
            text="Looks like you need to set up a kitchen"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              handleSubmit={[Function]}
              user={
                Object {
                  "email": "hass@mail.com",
                  "firstname": "Hasstru",
                  "kitchen": Object {
                    "MenuofTheDay": null,
                    "description": "This is my test menu",
                    "id": 3,
                    "name": "This is great",
                  },
                  "lastname": "Ezekiel",
                }
              }
            />,
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  "name",
                  "description",
                  "id",
                  "MenuofTheDay",
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-20vh",
                  "height": "10%",
                  "position": "relative",
                  "width": "100%",
                }
              }
              handleSubmit={[Function]}
              image="empty"
              kitchen={
                Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                }
              }
              meals={
                Array [
                  Object {
                    "description": "",
                    "kitchenId": 2,
                    "name": "This is a menu",
                    "price": 5000,
                  },
                ]
              }
              text="Looks like you need to set up a kitchen"
            />,
          ],
          "className": "main-profile-body",
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": undefined,
              "handleSubmit": [Function],
              "user": Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "callback": [Function],
              "determineRenderBy": Array [
                "name",
                "description",
                "id",
                "MenuofTheDay",
              ],
              "emptyContainerStyle": Object {
                "bottom": "-20vh",
                "height": "10%",
                "position": "relative",
                "width": "100%",
              },
              "handleSubmit": [Function],
              "image": "empty",
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "meals": Array [
                Object {
                  "description": "",
                  "kitchenId": 2,
                  "name": "This is a menu",
                  "price": 5000,
                },
              ],
              "text": "Looks like you need to set up a kitchen",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
